python/res/fm/ecl/ecl_config.py

```diff
@@ -59,68 +59,131 @@ class EclConfig(object):
     example file.
 
     """
-    DEFAULT_CONFIG_FILE = os.path.join(os.path.dirname(__file__), "ecl_config.yml")
 
-    def __init__(self):
-        config_file = os.getenv("ECL_SITE_CONFIG", default = self.DEFAULT_CONFIG_FILE)
+    def __init__(self, config_file):
         with open(config_file) as f:
             try:
                 config = yaml.load(f)
             except:
                 raise ValueError("Failed parse: {} as yaml".format(config_file))
 
         self._config = config
+        self._config_file = os.path.abspath(config_file)
+        self.default_version = None
```

Comment:
I think keeping information only once is a good thing, so how about making a property `default_version` that returns `self._config.get('default_version')` instead?

Reply:
Also, please consider defining the `default_version` somewhere.

Reply:
OK - I have added the property. Have also added a small "namespace" class `Keys` which has all the magic strings used.